src/ApplicationInsights.ServiceFabric.Native/Net45/RequestTrackingModule/ServiceRemotingServerEventListener.cs

```diff
@@ -0,0 +1,236 @@
+using System.Xml;
+using Microsoft.ApplicationInsights.Extensibility.Implementation.Tracing;
+using System.Collections.Concurrent;
+using Microsoft.ApplicationInsights.Extensibility.Implementation;
+
+namespace Microsoft.ApplicationInsights.ServiceFabric.Module
+{
+    internal class ServiceRemotingServerEventListener : IDisposable
+    {
+        private TelemetryClient client;
+        private TelemetryConfiguration configuration;
+        private string effectiveProfileQueryEndpoint;
+        private bool setComponentCorrelationHttpHeaders;
+        private readonly ICorrelationIdLookupHelper correlationIdLookupHelper;
+        private ConcurrentDictionary<IServiceRemotingRequestMessage, IOperationHolder<RequestTelemetry>> pendingTelemetry = new ConcurrentDictionary<IServiceRemotingRequestMessage, IOperationHolder<RequestTelemetry>>();
+
+        public ServiceRemotingServerEventListener(TelemetryConfiguration configuration, string effectiveProfileQueryEndpoint, bool setComponentCorrelationHttpHeaders, ICorrelationIdLookupHelper correlationIdLookupHelper = null)
+        {
+            this.client = new TelemetryClient(configuration);
+            this.client.Context.GetInternalContext().SdkVersion = SdkVersionUtils.GetSdkVersion("serviceremoting:");
+
+            this.configuration = configuration;
+            this.effectiveProfileQueryEndpoint = effectiveProfileQueryEndpoint;
+            this.setComponentCorrelationHttpHeaders = setComponentCorrelationHttpHeaders;
+            this.correlationIdLookupHelper = correlationIdLookupHelper ?? new CorrelationIdLookupHelper(effectiveProfileQueryEndpoint);
+
+            ServiceRemotingServiceEvents.ReceiveRequest += ServiceRemotingServiceEvents_ReceiveRequest;
+            ServiceRemotingServiceEvents.SendResponse += ServiceRemotingServiceEvents_SendResponse;
+        }
+
+        private void ServiceRemotingServiceEvents_ReceiveRequest(object sender, EventArgs e)
+        {
+            try
+            {
+                ServiceRemotingRequestEventArgs eventArgs = e as ServiceRemotingRequestEventArgs;
+
+                if (eventArgs == null)
+                {
+                    ServiceFabricSDKEventSource.Log.InvalidEventArgument((typeof(ServiceRemotingRequestEventArgs)).Name, e.GetType().Name);
+                    return;
+                }
+
+                var request = eventArgs.Request;
+                var messageHeaders = request?.GetHeader();
+
+                // If there are no header objects passed in, we don't do anything.
+                if (messageHeaders == null)
+                {
+                    ServiceFabricSDKEventSource.Log.HeadersNotFound();
+                    return;
+                }
+
+                string methodName = eventArgs.MethodName;
+
+                // Create and prepare activity and RequestTelemetry objects to track this request.
+                RequestTelemetry rt = new RequestTelemetry();
+                this.client.Initialize(rt);
```

Comment:
this potentially invokes dozen of initializers. StartOperation calls initalize. Moreover, StopOperation also calls it implicitly. Is it possible to remove this call and optimize a bit?